hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/RSProcedureDispatcher.java

```diff
@@ -371,6 +403,39 @@ private boolean isSaslError(IOException e) {
       }
     }
 
+    /**
+     * Returns true if the error or its cause is of type ConnectionClosedException.
+     * @param e IOException thrown by the underlying rpc framework.
+     * @return True if the error or its cause is of type ConnectionClosedException.
+     */
+    private boolean isConnectionClosedError(IOException e) {
+      if (e instanceof ConnectionClosedException) {
+        return true;
+      }
+      Throwable cause = e;
+      while (true) {
+        if (cause instanceof IOException) {
+          IOException unwrappedCause = unwrapException((IOException) cause);
+          if (unwrappedCause instanceof ConnectionClosedException) {
+            return true;
+          }
+        }
+        cause = cause.getCause();
+        if (cause == null) {
+          return false;
+        }
+      }
+    }
+
+    /**
+     * Returns true if the error is eligible for imposing retry limit.
+     * @param e IOException thrown by the underlying rpc framework.
+     * @return True if the error is eligible for imposing retry limit.
+     */
+    private boolean isErrorTypeEligibleForRetryLimit(IOException e) {
```

Comment:
Instead of "eligible for retry limit" type wording, you should describe this as "fast fail" in method naming and comment text.
Ok, and then following this recommendation, the config `hbase.master.rs.remote.proc.retry.limit` might be better named `hbase.master.rs.remote.proc.fast.fail.limit`